src/test/run-pass/issue-25579.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+enum Sexpression {
+  Num(()),
+  Cons(&'static mut Sexpression)
+}
+
+fn causes_ice(mut l: &mut Sexpression)
+{
```

Comment:
I have reformatted the code above, I think it makes sense to have it indented 4 spaces and the brace on the same line as the if statment, consistent with the rest of the file anyway. 

I need to resolve the failed build then hopefully this should be good to go.